src/modules/ekf2/EKF/aid_sources/gnss/gps_control.cpp

```diff
@@ -63,7 +63,8 @@ void Ekf::controlGpsFusion(const imuSample &imu_delayed)
 	if (_gps_data_ready) {
 		const gnssSample &gnss_sample = _gps_sample_delayed;
 
-		if (runGnssChecks(gnss_sample) && isTimedOut(_last_gps_fail_us, (uint64_t)_min_gps_health_time_us / 2)) {
+		if (runGnssChecks(gnss_sample)
+		    && isTimedOut(_last_gps_fail_us, max((uint64_t)1e6, (uint64_t)_min_gps_health_time_us / 10))) {
```

Comment:
```suggestion
		    ) {
```

Wouldn't this be fine?
 - GPS fusion won't start for the first time until all checks have passed continuously for _min_gps_health_time_us
 - if any GPS checks fail that sample won't be fused

Beyond this it could be enforced with more granularity if it actually matters.
eg GPS velocity fusion requiring `vel_var < _params.req_sacc`

Reply:
@bresch and I thought that it still can be useful to discard some measurements after a failed check since it's plausible that there are several unprecise measurement after the failed check.

What exactly do you mean? `runGnssChecks` is checking `vel_var < _params.req_sacc`

Reply:
> What exactly do you mean? runGnssChecks is checking vel_var < _params.req_sacc

I mean having the velocity related checks only apply to GNSS velocity fusion.

We can follow up and talk about a bigger split later.